src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java

```diff
@@ -90,8 +100,91 @@ private Filing generateAccountFiling() {
      * @return The location where the service has stored the generated ixbrl.
      */
     private String generateIxbrl() {
-        //TODO: call the document generator's new end point to get generate ixbrl. Implementation does NOT exist yet. (STORY SFA-574)
+        //TODO: call the document generator's new end point to get generate ixbrl. Implementation does NOT exist yet. (STORY SFA-595)
         return documentGeneratorCaller.generateIxbrl();
     }
 
+    /**
+     * Validates the ixbrl against TNEP. This validation is driven by the environment and it can be
+     * disable.
+     */
+    private boolean isValidIxbrl() {
+        //TODO: this will be set to true when TNEP Validation plugged in.(STORY SFA-574)
+        boolean isIxbrlValid = false;
+        if (!environmentReader.getMandatoryBoolean(DISABLE_IXBRL_VALIDATION_ENV_VAR)) {
+            //TODO Add TNEP validation to be added when functionality is implemented . (STORY SFA-574)
+            //isIxbrlValid will be set to false if fails the tnep validation.
+            isIxbrlValid =  true;
+        }
+
+        return isIxbrlValid;
+    }
+
+    /**
+     * Generates the filing based on the Filing model.
+     *
+     * @param transaction - transaction information
+     * @param accountsType - Account type information: account type, ixbrl's template name, account
+     * @param ixbrlLocation - the location where the ixbrl is stored.
+     * @return
+     */
+    private Filing createAccountFiling(Transaction transaction, AccountsType accountsType,
+        String ixbrlLocation) {
+        Filing filing = new Filing();
+
+        //TODO Set correct periodEndOn (it is the Current Period's end date" in mongodb). Waiting for Doc.Gen changes. (STORY SFA-595)
+        LocalDate periodEndDate = LocalDate.now();
+
+        filing.setCompanyNumber(transaction.getCompanyNumber());
+        filing.setDescriptionIdentifier(accountsType.getAccountType());
+        filing.setKind(accountsType.getKind());
+        filing.setDescriptionValues(getDescriptionValues(periodEndDate));
+        filing.setData(getFilingData(periodEndDate, ixbrlLocation));
+
+        return filing;
+    }
+
+    /**
+     * Get the description values, which currently contains the period end date. This data is
+     * required for the filing description.
+     *
+     * @param periodEndDate - account's period end date
+     * @return {@link Map<String,String>} containing period end date, to match filing model
+     */
+    private Map<String, String> getDescriptionValues(LocalDate periodEndDate) {
```

Comment:
Just to make you aware, the new document-generator will be returning the description values in the api response. So not sure you will need this anymore?

Reply:
I've done this change since the Document generator implementation has not been completed yet.  I will wait until the Document generator's implementation is completed before removing this method.  
There is already a story to plug in the document generator and the story number has been added to a "TODO" where  document-generator's information will be used. I will updated the "TODO"